core/src/trezor/wire/thp/pairing_context.py

```diff
@@ -0,0 +1,249 @@
+from typing import TYPE_CHECKING
+from ubinascii import hexlify
+
+import trezorui2
+from trezor import loop, protobuf, workflow
+from trezor.crypto import random
+from trezor.ui.layouts.tt import RustLayout
+from trezor.wire import context, message_handler, protocol_common
+from trezor.wire.context import UnexpectedMessageException
+from trezor.wire.errors import ActionCancelled
+from trezor.wire.protocol_common import Context, Message
+
+if TYPE_CHECKING:
+    from typing import Container
+
+    from .channel import Channel
+    from .cpace import Cpace
+
+    pass
+
+if __debug__:
+    from trezor import log
+
+
+class PairingDisplayData:
+    def __init__(self) -> None:
+        self.display_code_entry: bool = False
+        self.display_qr_code: bool = False
+        self.display_nfc_unidirectional: bool = False
+        self.code_code_entry: int | None = None
+        self.code_qr_code: bytes | None = None
+        self.code_nfc_unidirectional: bytes | None = None
+
+    def get_display_layout(self) -> RustLayout:
+        return RustLayout(
+            trezorui2.show_address_details(  # noqa
+                qr_title="Scan QR code to pair",
+                address=self._get_code_qr_code_str(),
+                case_sensitive=True,
+                details_title="",
+                account="Code to rewrite:\n" + self._get_code_code_entry_str(),
+                path="",
+                xpubs=[],
+            )
+        )
+
+    def _get_code_code_entry_str(self) -> str:
+        if self.display_code_entry and self.code_code_entry is not None:
+            code_str = str(self.code_code_entry)
+            print("code_code_entry:", code_str)
+
+            return code_str[:3] + " " + code_str[3:]
+        return "NOT ALLOWED"
```

Comment:
I would rise an exception here.

By the the way, having both `display_code_entry` and `code_code_entry` is redundant since `display_code_entry` is `True` if and only if `code_code_entry` is not `None`.

The same applies to `_get_code_qr_code_str`.

Reply:
Methods now raise exceptions if the relevant variable is None. The boolean values are discarded, as they are currently redundant. There can be a scenario, where we want to toggle the visibility of the displayed methods, but now those boolean values are not MVP.

Removal of boolean variables and added exceptions:
https://github.com/trezor/trezor-firmware/pull/4050/commits/bbb73f760ef2a472f78622fdd6bcff6ad076fd88